### ISSUE TITLE
Unable to upload

### ISSUE THREAD

**Issue by santinod** [HeyuX10Automation/heyu#2]
Today I upgraded from 2.6.0 to 2.10, and suddenly wasn't able to upload schedules any more to my CM11EFL.
Checking the various intermediate versions, I could find out that something got broken in 2.10-rc2 (2.10-rc1 was still OK).
Here's the log from `heyu -v upload`:

```
Version:2.10-rc2
Searching for '/home/santinod/.heyu/x10config'
Found configuration file '/home/santinod/.heyu/x10config'
Heyu directory /home/santinod/.heyu/ is writable.
Reading Heyu configuration file '/home/santinod/.heyu/x10config'
lockpid: Checking for file '/var/lock/LCK..heyu.write.ttyUSB0'
Trying to lock (/var/lock/LCK..heyu.write.ttyUSB0)
lockpid: Checking for file '/var/lock/LCK..heyu.write.ttyUSB0'
/var/lock/LCK..heyu.write.ttyUSB0 is locked
Alert acks are 0x52, 0x12, 0x42
lockpid: Checking for file '/var/lock/LCK..heyu.relay.ttyUSB0'
lockpid: Checking for file '/var/lock/LCK..heyu.relay.ttyUSB0'
There was already a relay running (pid = 9231)
Schedule: /home/santinod/.heyu/x10.sched
Entering replace_delayed_events() at generation 0
Entering combine_similar_events() at generation 0
Entering security_adjust_legal() at generation 1
Entering resolve_dates() at generation 1
Entering clear_disabled_events() at generation 2
Entering resolve_tevent_overlap_sep() at generation 3
Entering reconstruct_timers() at tevent generation 3
Entering resolve_sun_times() at timer generation 1
Expanded timers =  50
Max dawn/dusk error over the 366 day period = 0 minutes.
Interface memory free = 537 bytes ( = 52.4% )
See file /home/santinod/.heyu/report.txt for details.
xwrite() called, count=19
xread() called, count=1, timeout = 1
xread() returning 1 byte(s). The first is 0xf3
Checksum confirmed
xwrite() called, count=1
xread() called, count=1, timeout = 3
xread() returning 1 byte(s). The first is 0x55
Interface is ready.
Uploading 64 block memory image to interface.
Uploading block 63 to address 0x3f0.............................
xwrite() called, count=19
xread() called, count=1, timeout = 1
xread() returning 1 byte(s). The first is 0xe3
Checksum e3 NOT confirmed - should be 4e

Uploading block 63 to address 0x3f0
xwrite() called, count=19
xread() called, count=1, timeout = 1
xread() returning 1 byte(s). The first is 0x4e
Checksum confirmed
xwrite() called, count=1
xread() called, count=1, timeout = 3
Alarm - timeout
xread() returning 0 bytes
Uploading block 63 to address 0x3f0
xwrite() called, count=19
xread() called, count=1, timeout = 1
xread() returning 1 byte(s). The first is 0x4e
Checksum confirmed
xwrite() called, count=1
xread() called, count=1, timeout = 3
Alarm - timeout
xread() returning 0 bytes
Upload failed to write block 63 to address 0x3f0
```

**Comment by jkrzyszt** [comment]
Can you please check if the issue still exists for you in v2.11-rc2? I suspect it might be caused by a configuration setting not compatible with your platform and that is supposed to have been fixed automagically since we switched to Autotools generated settings for v2.11.

Thanks,
Janusz

PS. Sorry for very late response, I've just realised notifications from GitHub haven't been reaching me.

**Comment by santinod** [comment]
Looks like you're right, the issue is gone in 2.11-rc2. Thanks!

**Comment by santinod** [comment]
Still no joy...

```
Version:2.11-rc2
Searching for '/home/santinod/.heyu/x10config'
Found configuration file '/home/santinod/.heyu/x10config'
Heyu directory /home/santinod/.heyu/ is writable.
Reading Heyu configuration file '/home/santinod/.heyu/x10config'
lockpid: Checking for file '/var/lock/LCK..heyu.write.ttyUSB0'
Trying to lock (/var/lock/LCK..heyu.write.ttyUSB0)
lockpid: Checking for file '/var/lock/LCK..heyu.write.ttyUSB0'
/var/lock/LCK..heyu.write.ttyUSB0 is locked
Alert acks are 0x52, 0x12, 0x42
lockpid: Checking for file '/var/lock/LCK..heyu.relay.ttyUSB0'
lockpid: Checking for file '/var/lock/LCK..heyu.relay.ttyUSB0'
There was already a relay running (pid = 21310)
Schedule: /home/santinod/.heyu/x10.sched
Entering replace_delayed_events() at generation 0
Entering combine_similar_events() at generation 0
Entering security_adjust_legal() at generation 1
Entering resolve_dates() at generation 1
Entering clear_disabled_events() at generation 2
Entering resolve_tevent_overlap_sep() at generation 3
Entering reconstruct_timers() at tevent generation 3
Entering resolve_sun_times() at timer generation 1
Expanded timers =  40
Max dawn/dusk error over the 366 day period = 0 minutes.
Interface memory free = 627 bytes ( = 61.2% )
See file /home/santinod/.heyu/report.txt for details.
xwrite() called, count=19
xread() called, count=1, timeout = 1
xread() returning 1 byte(s). The first is 0xf3
Checksum confirmed
xwrite() called, count=1
xread() called, count=1, timeout = 3
Alarm - timeout
xread() returning 0 bytes
upload_eeprom_image() failed to erase initiator block.
```

**Comment by jkrzyszt** [comment]
Assuming v2.10-rc1 still works for you, please post the debug output of the working `heyu -v upload` for comparison.

**Comment by jkrzyszt** [comment]
No response from user, closing.

**Comment by zonedgopher** [comment]
I realize this thread is very old, however I am posting my experience using fedora v29 on intel nuc i7 platform
[heyu-upload-v294.txt](https://github.com/HeyuX10Automation/heyu/files/4606064/heyu-upload-v294.txt)
 for the same issue as above - the heyu upload for cm11a fails.  It gives an error "failed to erase initiator block".
Version 2.9.1        works perfectly.
version 2.94         works perfectly.
version 2.10-rc1   works perfectly
version 2.10-rc2   fails on heyu upload
version 2.10         fails on heyu upload
version 2.10.1       fails on heyu upload
version 2.11-rc3    fails on heyu upload.
Any dat you would like collected please let me know.  Attached are the heyu -v upload files for version 2.9.4 that works and version 2.11-rc3 that fails.
[heyu-upload-v211rc3.txt](https://github.com/HeyuX10Automation/heyu/files/4606065/heyu-upload-v211rc3.txt)
[heyu-upload-v294.txt](https://github.com/HeyuX10Automation/heyu/files/4606067/heyu-upload-v294.txt)

**Comment by jkrzyszt** [comment]
Resolved in [Heyu 2.10.2](https://github.com/HeyuX10Automation/heyu/releases/tag/v2.10.2)+, see #22.  Still not resolved in 2.11, progress, if any, tracked under #30.
